csharp/extractor/Semmle.Extraction/Id.cs

```diff
@@ -34,7 +34,12 @@ public class FreshId : IId
 
         public override string ToString() => "*";
 
-        public override bool Equals(object obj) => obj.GetType() == GetType();
+        public override bool Equals(object? obj)
+        {
+            // Expand logic to allow for nullability control flow analysis
```

Comment:
What happens if you do `public override bool Equals(object obj) => obj?.GetType() == GetType();` instead?

Reply:
Should be the same, so let's do that.